From: Petr Tesarik <ptesarik@suse.com>
Date: Fri, 3 Apr 2020 13:12:00 +0200
Subject: kexec-tools: s390: Reset kernel command line on syscall fallback
References: bsc#1167868
Upstream: merged
Git-commit: 5dc72ded6050a5bb161e3b832867e660c9ca009e

The command line is duplicated on s390 if kexec_file_load(2) is not
implemented. That's because the corresponding variable is not reset
to an empty string before re-parsing the kexec command line.

Fixes: 9cf721279f6c ("Reset getopt before falling back to legacy syscall")
Signed-off-by: Petr Tesarik <ptesarik@suse.com>
---
 kexec/arch/s390/kexec-image.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/kexec/arch/s390/kexec-image.c b/kexec/arch/s390/kexec-image.c
index 8b39566..3c24fdf 100644
--- a/kexec/arch/s390/kexec-image.c
+++ b/kexec/arch/s390/kexec-image.c
@@ -112,6 +112,7 @@ image_s390_load(int argc, char **argv, const char *kernel_buf,
 		};
 	static const char short_options[] = KEXEC_OPT_STR "";
 
+	command_line[0] = 0;
 	ramdisk = NULL;
 	ramdisk_len = 0;
 	ramdisk_origin = 0;
-- 
2.16.4

